### DISCUSSION TITLE
Sprite ColorDepth

### DISCUSSION THREAD

**Discussion by greglogan** [Bodmer/TFT_eSPI#2930] [Q&A - Sprites]
Hi 
I created a series of sketches back over a year ago - it was a series of gauges for an old car. Everything was working fine but there was some issues with the math calcs for sensor readings. No big deal, just a bit of tweaking. After almost a year, I decided to implement these changes. When I uploaded the amended sketches back to my ESP32, the sketches won't run. After a lot of checking, it 'seems' like there is a new issue - I can only get the sketches to run when setColorDepth is set to 8 - previously this worked just fine set at 16. So it all runs fine now, but the main image on the sprite is only working on 8 bit colours which makes it all look a bit crap.

I probably have made some sort of stupid change, but I can't find where. Is there any help anyone can give? Code affected is below - when set to 8, it'll load and run fine, when changed to 16, the ESP32 and GC9A01 just lock up

void createNeedle(){
  needle.setColorDepth(16);
  needle.createSprite(23,91);
  needle.pushImage(0,0,23,91, NEEDLE);
  needle.setPivot(11,79);

**Comment by Bodmer** [comment]
This indicates the RAM available has changed for some reason.

Espressif seem to keep tweaking the memory partitioning in the board package and introducing new unsolved issues, so try reverting the board package to the one you used over 1 year ago. I think this suggests a lack of proper software management and control within the Espressif software development team!

Don't upgrade TFT_eSPI either as it may not be compatible anymore with old pre 2.0.x board packages as I have to keep tweaking it in line with the later board package changes!

**Reply by greglogan** [reply]
that's a great call - I suspected it might have been a library issue and hadn't thought about boards. Issue resolved! thank you
